### ISSUE TITLE
Hybrid iOS app - Launching Expensify page displayed after close account

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51051]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.50
**Reproducible in staging?:** Y
**Reproducible in production?:**Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** No
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5094469
**Email or phone of affected tester (no customers):** rybkina+101724y@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Login as new user to Hybrid iOS app
2. Go to Setting > Security > Close account
3. Enter account email
4. Tap "Close account"

## Expected Result:
Account should be close. Sign in page should displayed

## Actual Result:
Launching Expensify page displayed after close account

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/f4a96a3f-f029-4ff8-b24a-16529553b3d9

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@sonialiap  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by sonialiap** [comment]
Looking for proposals

**Comment by sonialiap** [comment]
You're right! Changing to internal

**Comment by sonialiap** [comment]
booping melvin

**Comment by sonialiap** [comment]
Since this is low and we need an internal engineer, moving to weekly

**Comment by sonialiap** [comment]
Looking for an internal engineer

**Comment by sonialiap** [comment]
low priority, adding retest label

**Comment by sonialiap** [comment]
[Asked for a retest](https://expensify.slack.com/archives/C9YU7BX5M/p1739954557401909)

**Comment by izarutskaya** [comment]
Issue is still reproducible 

https://github.com/user-attachments/assets/f56fd9c0-5ab6-4475-812e-702465222d4f

**Comment by carlosmiceli** [comment]
@sonialiap I did some triaging, but I think maybe someone from the SW team can help us here (maybe @mateuuszzzzz)? I can see that the account is closed successfully, so I'm leaning towards a FE fix. Moved it to weekly because it's a pretty easy bug for users to see.

**Comment by carlosmiceli** [comment]
Just asked for help in the SW channel.

**Comment by carlosmiceli** [comment]
Thanks sir! All yours!

**Comment by mateuuszzzzz** [comment]
I investigated this issue today. It will be hard to guarantee smooth transition to OldDot sign-in page. Here's why:

When we call the `closeAccount` command, we clear Onyx implicitly via OnyxUpdates, which causes the navigators to swap and display the expired session page. Even if we close NewDot app immediately after the `closeAccount` request finishes, this page is still briefly visible which leads to poor user experience.

We could work around this issue by intercepting the Onyx clear operation in `OnyxUpdates`. Before clearing the storage, we would close the NewDot modal and then terminate the React Native instance when storage is empty. However, this approach is somewhat hacky.

On the other hand, we have this PR https://github.com/Expensify/Mobile-Expensify/pull/13284 which replaces the sign-in page in HybridApp with NewDot's version instead of OldDot's.

Using the NewDot sign-in page would eliminate the issue. However... this would reverse the situation, meaning we'd need to implement the same mechanism in OldDot. Specifically, if someone closes their account on OldDot, we should ensure they see the NewDot sign-in page instead. Since OldDot is much less complex, implementing this solution there would be significantly easier

**Comment by carlosmiceli** [comment]
> if someone closes their account on OldDot, we should ensure they see the NewDot sign-in page instead

Sorry, not sure I understand why this is the case, why can't they see the OldDot sign-in page if they close their OldDot account?

**Comment by mateuuszzzzz** [comment]
Oh, sorry! My point was that while we can implement it this way, this approach also comes with some difficult-to-overcome issues. Specifically, the expired session page briefly flickers due to the command clearing Onyx. Intercepting this process is very challenging because it results from an Onyx update received in the command's API response. This isn't an issue on non-hybrid platforms, but it seems to cause difficulties on HybridApp.

Nevertheless, I can try to find out a workaround for this

**Comment by carlosmiceli** [comment]
Gotcha, let me ask about this and see what we'd rather do! Will report back 👍

**Comment by carlosmiceli** [comment]
@mateuuszzzzz sorry to keep asking, still trying to understand some details about how HybridApp works... How can we implement the flow you suggest (closing the account in OldDot -> seeing the NewDot sign-in page) only in HybridApp? We don't want users in other platforms to close their OldDot account and be taken to new.expensify.com, but if we can keep it to HybridApp then that's great. Do we have any other cases where we do something like this?

**Comment by mateuuszzzzz** [comment]
On HybridApp we have 2 independent apps. Old app and new app, but they share single sign-in page (the one from OldDot). My suggestion simply changes this and we would use NewDot's sign-in page. And yes, we can keep it entirely to HybridApp 😅

But this [PR](https://github.com/Expensify/Mobile-Expensify/pull/13284) I mentioned still requires some backend changes to make it work correctly.


I also have one alternative idea:

No matter how quick we navigate back to OldDot on account closure we still see expired session page. I explained above that this happens due to Onyx updates included in `CloseAccount` response.

I wonder if it is possible to stop sending clear onyx update in response of `CloseAccount` (but only on HybridApp).
Then we could:

1. Hide NewDot app and display OldDot's sign-in page (via native method)
2. Clear Onyx in the background (we would need to trigger it manually)
3. Kill React Native instance once it is not needed (via native method)

Another alternative is to intercept this Onyx update directly in the middleware that handles API command responses. This would allow us to apply the fix directly on the frontend, but it's a bit hacky since we would need to control UI  in middleware (hiding ND and displaying OD sign-in page).

**Comment by carlosmiceli** [comment]
Let's skip the hacky option and get it right ;)

From what you're saying, it still feels like the first option is the cleanest one (and the one that follows the already agreed design decision based on that PR), right?

**Comment by mateuuszzzzz** [comment]
Yes 😅 I think the backend part of this PR is already being worked on. Once it's ready we can continue work on this issue

**Comment by carlosmiceli** [comment]
Sounds good! Let's revisit once that's complete.

**Comment by carlosmiceli** [comment]
Hi @mateuuszzzzz, do you know if there's any updates on this? Thank you!

**Comment by mateuuszzzzz** [comment]
Hi! We're unblocked with `SignInPage` PR. @289Adam289 is working on it

**Comment by carlosmiceli** [comment]
@mateuuszzzzz @289Adam289 howdy! What's the status of the PR?

**Comment by mateuuszzzzz** [comment]
`SignInPage` PR has now high priority to finish it before F1 movie. 

Currently we have 2 issues left:
1. W need to verify that SignIn with Google and Apple works 
2. We need to wait for backend changes, afaik @Julesssss is working on them

**Comment by mateuuszzzzz** [comment]
It turned out that new `SignInPage` handles closing account pretty well without any changes so we just need to wait for `SignInPage` and then this issue can be closed. 🎉

**Comment by sonialiap** [comment]
@carlosmiceli I'm OOO June 2-13 but since it looks like we'll be closing this issue I'm not going to add a BZ leave buddy. If you do need a BZ please reapply the bug label for a new assignee
